app/src/main/java/com/group80/uoftinder/GenerateCompatibilityList.java

```diff
@@ -0,0 +1,113 @@
+package com.group80.uoftinder;
+
+import com.group80.uoftinder.entities.User;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Hashtable;
+import java.util.List;
+
+/**
+ * A use case that generates a list of compatible users for the current user
+ */
+public class GenerateCompatibilityList {
+    private User curUser;
+    private int curUserScore;
+    private List<String> compatibilityList;
+    private RecPresenterInterface recPresenterInterface;
+    private List<User> allUsers;
+    private UserScoreFacade usf;
+
+    /**
+     * Initialize the class by loading in all the users in the database, creating the
+     * UserScoreFacade instance, and getting the score of curUser
+     */
+    public GenerateCompatibilityList(RecPresenterInterface recPresenterInterface) {
+        List<User> allUsers = UserRealtimeDbFacade.getAllUsers(curUser.getUserType());
+        usf = new UserScoreFacade(curUser, new CreateAccountInteractor()); // may need to change 2nd param
+        User curUser = UserRealtimeDbFacade.getCurrentUser();
+        int curUserScore = curUser.getScore();
+        this.recPresenterInterface = recPresenterInterface;
+    }
+
+    /**
+     * Generate the compatibility list for curUser and update the User's compatibilityList attribute.
+     * Pass the compatibilityList to recPresenterInterface so that it can display it
+     */
+    public void generateCompatibilityList() {
+        compatibilityList = new ArrayList<>();
+        Map<String, Integer> compScores = calculateCompatibilityScores(allUsers);
+        for (int i = 0 ; i < users.size() ; i++) {
```

Comment:
From what I understand, you're currently using selection sort to arrange a list of compatible users by retrieving the next greatest compatibility score one-by-one. This requires writing multiple methods and is not very efficient on a large scale. Have you considered using the in-built ArrayList sort method using a Comparator to sort User objects by their compatibility score?

I found this page to be helpful: https://stackoverflow.com/questions/61205215/how-to-sort-an-arraylist-using-a-map-key-as-list-object-and-value-as-the-order

In your case, you might be able to try the following (assuming compScores is a Map<User, Integer>):

1. Define compatibilityList as an ArrayList of User objects
2. Add every user in allUsers to compatibilityList
3. Define a Comparator for User objects by doing the following: ```Comparator<User> userScoreComparator = Comparator.comparing(user -> compScores.get(user));```
4. Sort your list: ```compatibilityList.sort(userScoreComparator);```

Let me know if this makes sense!

Reply:
I am actually not entirely sure how the Comparator class for the users works. Let's discuss this sometime so that my implementation is more efficient.

Reply:
Hopefully our conversation helped! Let me know if you have any more questions about how this works.